bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/service/BusinessPartnerMappings.kt

```diff
@@ -65,7 +65,7 @@ class BusinessPartnerMappings {
                     } }
                     .map(::toIdentifierDto)
             }?: emptyList(),
-            states = entity.states.map(::toStateDto),
+            states = toStateDtos(entity.states, BusinessPartnerType.GENERIC),
```

Comment:
I'm just looking at the identifiers behaviour. Shouldn't the states not better mirror the behaviour of the identifiers and take the states based on the address type?

Reply:
In my opinion, yes we can do that and it will be right approach as well.